torch/distributed/_tensor/_collective_utils.py

```diff
@@ -158,3 +160,139 @@ def mesh_all_to_all(
             async_op=async_op,
         )
     return work
+
+
+def spec_to_bytes(spec: "placement_types.DTensorSpec") -> int:
+    assert spec.tensor_meta is not None, "spec should have tensor meta defined!"
+    dtype_to_bytes = {
```

Comment:
You can get `tensor.dtype.itemsize`

Reply:
cool good to know there's already a API for it, I'll update